libflux/Makefile

```diff
@@ -6,9 +6,12 @@ build:
 	wasm-pack build --dev --scope influxdata
 
 generate:
+	flatc --rust -o src/ast/ ../ast/ast.fbs
```

Comment:
Should we integrate this into `build.rs`? I think the makefile is probably fine since I don't think it is necessary to require `flatc` be present to build the Rust module though.

Can we get this with proper Makefile targets and sources though? I would also prefer if we kept `ast_generated.rs` and did not publicly expose the module. This would simplify the makefile and I don't think the specific name matters since the module wouldn't be publicly exposed anyway.

Reply:
Hmm, I've some questions about this.  I didn't know about `build.rs`.

So the lexer is generated as part of `cargo build`, then?  Does that mean if `scanner.rl` is out of sync with its generated code, that `make checkgenerate` will not catch it?

(I was also a little surprised to see that there are two identical copies of `scanner.rl` in our repo)

I'm fine with generating the code from the Makefile or from `build.rs`, but it seems like the mechanism we use to do it should be the same as for the lexer.

Reply:
For the Rust code, the scanner is not checked in and is constructed with `build.rs`. I am wondering if this is the correct way to do it though. It assumes that everyone who uses the code and performs the build will also need these same tools. So if you want to build the Rust parser for Flux inside of InfluxDB, then the InfluxDB build environment will need these tools present to generate the code.

I do agree that we should do the same thing for both of these. Maybe we should convert the scanner to generate the code as part of the makefile and commit the generated code just like we do with Go.

Reply:
> I would also prefer if we kept ast_generated.rs and did not publicly expose the module.

The generated code is not publicly exposed now, as it's nested within the `ast` module and doesn't have the keyword `pub`.  

My understanding of Rust modules is that if I left the file with its original name of `ast_generated.rs`, then the module would have to be named `ast_generated` which doesn't convey very well what it is.  I think maybe I'm misunderstanding your question though, we should talk tomorrow.

Reply:
Ok. I think this is fine.

Reply:
One caveat. Can you move the `mv` to the same line and do `&& mv ...`?